compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -777,10 +777,18 @@ OMR::ARM64::TreeEvaluator::BBStartEvaluator(TR::Node *node, TR::CodeGenerator *c
       child->decReferenceCount();
       }
 
+   TR::Instruction *labelInst = NULL;
    if (node->getLabel() != NULL)
       {
-      node->getLabel()->setInstruction(generateLabelInstruction(cg, TR::InstOpCode::label, node, node->getLabel(), deps));
+      node->getLabel()->setInstruction(labelInst = generateLabelInstruction(cg, TR::InstOpCode::label, node, node->getLabel(), deps));
       }
+   else
+      {
+      TR::LabelSymbol *label = generateLabelSymbol(cg);
+      node->setLabel(label);
+      node->getLabel()->setInstruction(labelInst = generateLabelInstruction(cg, TR::InstOpCode::label, node, label, deps));
+      }
+   block->setFirstInstruction(labelInst);
```

Comment:
I think you could rewrite this like this to save some code space:
```
   TR::LabelSymbol *labelSym = node->getLabel();
   if (!labelSym)
      {
      labelSym = generateLabelSymbol(cg);
      node->setLabel(labelSym);
      }

   TR::Instruction *labelInst = generateLabelInstruction(cg, TR::InstOpCode::label, node, labelSym, deps);
   labelSym->setInstruction(labelInst);
   block->setFirstInstruction(labelInst);
```

Reply:
I rewrote this to make it more concise as you suggested